bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/discovery/NetatmoModuleDiscoveryService.java

```diff
@@ -195,4 +202,46 @@ private ThingUID findThingUID(String thingType, String thingId) throws IllegalAr
 
         throw new IllegalArgumentException("Unsupported device type discovered : " + thingType);
     }
+
+    private String createWeatherStationName(NAMain station, boolean isFavorite) {
+        StringBuilder nameBuilder = new StringBuilder();
+        nameBuilder.append(localizeType(station.getType()));
+        if(station.getStationName() != null) {
+            nameBuilder.append(' ');
+            nameBuilder.append(station.getStationName());
+        }
+        if(isFavorite) {
+            nameBuilder.append(" (favorite)");
+        }
+        return nameBuilder.toString();
+    }
+
+    private String createWeatherModuleName(NAMain station, NAStationModule module, boolean isFavorite) {
+        StringBuilder nameBuilder = new StringBuilder();
+        if(module.getModuleName() != null) {
+            nameBuilder.append(module.getModuleName());
+        } else {
+            nameBuilder.append(localizeType(module.getType()));
+        }
+        if(station.getStationName() != null) {
+            nameBuilder.append(' ');
+            nameBuilder.append(station.getStationName());
+        }
+        if(isFavorite) {
+            nameBuilder.append(" (favorite)");
+        }
+        return nameBuilder.toString();
+    }
+
+    private String localizeType(String typeName) {
```

Comment:
I think the core framework already provides a method for that. I have to search.

Reply:
It is described here and I found usages/examples within other bundles: https://www.openhab.org/docs/developer/utils/i18n.html

But I'm also wondering why it is so complicated to translate a String... It would be neat if the core framework would provide a shorter / easier to use solution.